### ISSUE TITLE
Do latents need to be scaled by the scheduler first?

### ISSUE THREAD

**Issue by off99555** [huggingface/diffusers#3544]
https://github.com/huggingface/diffusers/blob/c13dbd5c3a53017d27de35ad77b8d57f04c8ec7c/src/diffusers/pipelines/controlnet/pipeline_controlnet.py#L952-L963

I'm not sure if it's a bug but `latents` that is mentioned inside the if-block isn't scaled using `scheduler.scale_model_input()` unlike `latent_model_input`.

**Comment by patrickvonplaten** [comment]
Amazing find @off99555 ! You are 100% correct, this was a bug and thanks to you is now fixed in https://github.com/huggingface/diffusers/pull/3571

This wasn't noticed before because the default schedulers don't scale the model inputs. However euler and co. schedulers do. So the following works now:

```py
#!/usr/bin/env python3
import torch
import os
from huggingface_hub import HfApi
from pathlib import Path
from diffusers.utils import load_image
from controlnet_aux import ContentShuffleDetector

from diffusers import (
    ControlNetModel,
    EulerDiscreteScheduler,
    EulerDiscreteScheduler,
    StableDiffusionControlNetPipeline,
    UniPCMultistepScheduler,
)
import sys

checkpoint = sys.argv[1]

url = "https://github.com/lllyasviel/ControlNet-v1-1-nightly/raw/main/test_imgs/city.jpg"
image = load_image(url)

prompt = "New York"


processor = ContentShuffleDetector()
image = processor(image)

controlnet = ControlNetModel.from_pretrained(checkpoint, torch_dtype=torch.float16)
pipe = StableDiffusionControlNetPipeline.from_pretrained(
    "runwayml/stable-diffusion-v1-5", controlnet=controlnet, torch_dtype=torch.float16
)

# pipe.scheduler = UniPCMultistepScheduler.from_config(pipe.scheduler.config)
pipe.scheduler = EulerDiscreteScheduler.from_config(pipe.scheduler.config)
pipe.enable_model_cpu_offload()

generator = torch.manual_seed(33)
out_image = pipe(prompt, num_inference_steps=20, generator=generator, image=image).images[0]

path = os.path.join(Path.home(), "images", "aa.png")
out_image.save(path)

api = HfApi()

api.upload_file(
    path_or_fileobj=path,
    path_in_repo=path.split("/")[-1],
    repo_id="patrickvonplaten/images",
    repo_type="dataset",
)
print("https://huggingface.co/datasets/patrickvonplaten/images/blob/main/aa.png")
```

**Comment by off99555** [comment]
Cool. Works!
